python/test/unit/io/test_XDMF.py

```diff
@@ -165,7 +178,8 @@ def test_save_1d_scalar(tempdir, encoding):
     # FIXME: This randomly hangs in parallel
     V = FunctionSpace(mesh, "Lagrange", 2)
     u = Function(V)
-    u.vector()[:] = 1.0
+    A = imaginary_part()
```

Comment:
Seems too long and cryptic to me. What about `u.vector()[:] = 1.0 + 1j * has_petsc_complex()` ?

Reply:
`1.0 + 1j * has_petsc_complex()` returns a complex number with zero imaginary part `(1+0j)`.
However, in real mode a float is expected.

Reply:
It could also be replaced by:
```
A = 1j if has_petsc_complex() else 0.
u.vector()[:] = 1.0 + A
 ```
or something like.

Reply:
Ok, I would replace `A` with `i`. Its not too important anyway, these are just tests.

Reply:
Let's make it:
```
u.vector()[:] = 1.0 + (1j if has_petsc_complex() else 0)
```
This is perfectly readable on non-cryptic.